pipeline-adapters/mlrun-pipelines-kfp-common/src/mlrun_pipelines/common/ops.py

```diff
@@ -680,3 +684,121 @@ def _enrich_node_selector(function):
         project_node_selector, function_node_selector
     )
     return mlrun.utils.helpers.to_non_empty_values_dict(function_node_selector)
+
+
+def enrich_kfp_workflow_credentials(
+    byte_buffer: bytes,
+    content_type: str,
+    env_vars: list[V1EnvVar],
+) -> bytes:
+    if content_type.endswith("zip"):
+        modified_zip_bytes = _enrich_kfp_workflow_zip_credentials(
+            yaml_bytes=byte_buffer,
+            env_vars=env_vars,
+        )
+        return modified_zip_bytes
```

Comment:
this looks like some cpu-bound operation. could that be running aside from the api-serving process?